task/hiffy/src/common.rs

```diff
@@ -1205,3 +1257,53 @@ pub(crate) fn block_size(
 
     Ok(4)
 }
+
+// intent, auth_len, auth_data(Lease)
+#[cfg(feature = "update")]
+pub(crate) fn reset_component(
+    stack: &[Option<u32>],
+    data: &[u8],
+    _rval: &mut [u8],
+) -> Result<usize, Failure> {
+    if stack.len() < 3 {
```

Comment:
This appears to be an almost complete copy-paste of the previous function, down to a copy-paste of the typo on line 1290. Please extract a function describing the common bits so that it's clear that they're intended to be the same and to ensure they don't diverge.